lib/AArch64Base.ml

```diff
@@ -1385,27 +1381,15 @@ let do_pp_instruction m =
     pp_vreg v r2 ^ "," ^
     do_pp_idx true ra idx in
 
-  let pp_smem memo v rt ra kr =
+  let pp_smem memo v rt ra idx =
     pp_memo memo ^ " " ^ pp_vsimdreg v rt ^
-    ",[" ^ pp_xreg ra ^ pp_kr false false kr ^ "]" in
+    "," ^ pp_addr ra idx in
 
-  let pp_smem_shift memo v rt ra kr s =
-    pp_memo memo ^ " " ^ pp_vsimdreg v rt ^
-    ",[" ^ pp_xreg ra ^ pp_kr false false kr ^
-    pp_barrel_shift "," s (m.pp_k) ^ "]" in
-
-  let pp_smem_post memo v rt ra k =
-    pp_memo memo ^ " " ^ pp_vsimdreg v rt ^
-    ",[" ^ pp_xreg ra ^ "]" ^ m.pp_k k in
-
-  let pp_smemp_post memo v r1 r2 ra k =
-    pp_memo memo ^ " " ^ pp_vsimdreg v r1 ^ "," ^ pp_vsimdreg v r2 ^
-    ",[" ^ pp_xreg ra ^ "]" ^ m.pp_k k in
-
-  let pp_smemp memo v r1 r2 ra k =
-    pp_memo memo ^ " " ^ pp_vsimdreg v r1 ^ "," ^ pp_vsimdreg v r2 ^
-    ",[" ^ pp_xreg ra ^ (if m.compat
-      then pp_kr false false (K k) else "," ^ m.pp_k k)  ^ "]" in
+  let pp_smemp memo v r1 r2 ra idx =
+    pp_memo memo ^ " " ^
+    pp_vsimdreg v r1 ^ "," ^
+    pp_vsimdreg v r2 ^ "," ^
+    do_pp_idx true ra idx in
```

Comment:
This is getting a bit complicated but if you made the change below you wouldn't have to change the hashes of tests with LDP/STP. I think this means that we need to change the declaration of `do_pp_idx` and make sure that the argument from_pair is set to true for non-neon pair instructions only.

```suggestion
    do_pp_idx false ra idx in
```